ppl/archive/stat.go

```diff
@@ -169,13 +165,19 @@ func RecordCount(ctx context.Context, ark *Archive) (uint64, error) {
 }
 
 func Stat(ctx context.Context, zctx *resolver.Context, ark *Archive) (zbuf.ReadCloser, error) {
+	defs, err := ark.ReadDefinitions(ctx)
+	if err != nil {
+		return nil, err
+	}
 	ctx, cancel := context.WithCancel(ctx)
 	s := &statReadCloser{
-		ctx:    ctx,
-		cancel: cancel,
-		ark:    ark,
-		zctx:   zctx,
-		recs:   make(chan *zng.Record),
+		ctx:      ctx,
+		cancel:   cancel,
+		defmap:   defs.Map(),
```

Comment:
I don't understand why these `defmap` and `defstats` (and associated functions) are necessary?

Reply:
`indexStat` is a bad name but for the stats command I found it useful for the stats records to include a human-friendly name for each index. This also helped simplify tests as I didn't have to work about substituting ksuids.